controllers/spiaccesstokenbinding_controller.go

```diff
@@ -139,6 +177,23 @@ func (r *SPIAccessTokenBindingReconciler) Reconcile(ctx context.Context, req ctr
 	lg = lg.WithValues("linked_to", binding.Status.LinkedAccessTokenName,
 		"phase_at_reconcile_start", binding.Status.Phase)
 
+	finalizationResult, err := r.finalizers.Finalize(ctx, &binding)
+	if err != nil {
+		// if the finalization fails, the finalizer stays in place, and so we don't want any repeated attempts until
+		// we get another reconciliation due to cluster state change
+		return ctrl.Result{Requeue: false}, fmt.Errorf("failed to finalize: %w", err)
```

Comment:
you don't have to explicitly set `Requeue: false`, but nothing wrong with that...